dash_docs/chapters/dash_enterprise/dash_enterprise_chapters.py

```diff
@@ -3062,55 +3066,297 @@ def snapshot_page(n_clicks, url, wait_selector):
+        =====> <your-dash-app> web container output:
+        [...]
+        Failed to find application object 'server' in 'index'
+        [...]
+            raise HaltServer(reason, self.APP_LOAD_ERROR)
+            gunicorn.errors.HaltServer: <HaltServer 'App failed to load.' 4>
+        [...]
+                raise HaltServer(reason, self.APP_LOAD_ERROR)
+            gunicorn.errors.HaltServer: <HaltServer 'App failed to load.' 4>
+            Failed to find application object 'server' in 'index'
+            [2020-11-02 18:17:22 +0000] [191] [INFO] Worker exiting (pid: 191)
+        =====> end aa-qa web container output
+        To <your-dash-enterprise-host>:<your-dash-app>
+        ! [remote rejected] master -> master (pre-receive hook declined)
+        error: failed to push some refs to 'dokku@<your-dash-enterprise-host>:<your-dash-app>'
+        ```        
+        '''
+    ),
+
+    rc.Markdown(
+        '''
+        This happens when `server = app.server` is missing from your dash app .\
+        See  <dccLink href="/dash-enterprise/application-structure" children="Application Structure"/>
+        page for more details.\
+        '''
+    ),
+
+
+    rc.Markdown(
+        '''
+        ### Unable to select a buildpack
+        '''
+    ),
+
+    rc.Markdown(
+        '''    
+        ```shell
+        $ git push plotly master
+        [...]
+        -----> Adding BUILD_ENV to build environment...
+        -----> Unable to select a buildpack
+        To <your-dash-enterprise-host>:<your-dash-app>
+        ! [remote rejected] master -> master (pre-receive hook declined)
+        error: failed to push some refs to '<your-dash-enterprise-host>:<your-dash-app>'
         ```
-        ''',
-        style=styles.code_container),
+        '''
+    ),
 
-        rc.Markdown(
+    rc.Markdown(
+        '''
+
+        This error might occur if you are trying to push from a branch
+        that is not your `master` branch.\
+
+        To resolve get the name of your current
+        branch by running `git branch`. Then, to push from this branch
+        to the remote server, run `git push plotly your-branch-name:master`.\
+
+        This may also occur if you don’t have a requirements.txt file.\
+
+        The buildpack is the technology Dash Enterprise uses to create a Docker container.\
+
+        Dash Enterprise supports three different buildpacks: The standard Python & pip buildpack, a Python & conda buildpack, and a Dash for R buildpack. Dash Enterprise automatically selects the buildpack based off of which files are contained in the project:\
+
+        - `requirements.txt` - If this file exists, then Dash Enterprise uses the Python & pip buildpack.\
+        - `conda-requirements.txt` - If this file exists, then Dash Enterprises uses the Python & conda buildpack.\
+        - `.buildpacks` - If this file exists and is equal to `https://github.com/plotly/heroku-buildpack-r#heroku-18`, then Dash Enterprise uses the R buildpack.\
```

Comment:
```suggestion
        - `requirements.txt` - If this file exists, then Dash Enterprise uses the Python & pip buildpack.\
        
        - `conda-requirements.txt` - If this file exists, then Dash Enterprises uses the Python & conda buildpack.\
        
        - `.buildpacks` - If this file exists and is equal to `https://github.com/plotly/heroku-buildpack-r#heroku-18`, then Dash Enterprise uses the R buildpack.\
```

![image](https://user-images.githubusercontent.com/1280389/100395681-ca344d00-300f-11eb-88d9-128b4cf63a98.png)